DOCS-1247-PGD-6.0-Docs (#6763)
EnterpriseDB/docs: product_docs/docs/pgd/6/reference/node_management/node_uuids.mdx

---
title: Node UUIDs
navTitle: Node UUIDs
description: Node UUIDs in PGD
---

In PGD 6, each node now has a UUID that is used to identify the node in the cluster. This UUID is generated when the node is created and is unique to that node. The UUID can be found in various places in PGD, including:

* The `bdr.node` table, which contains information about each node in the cluster.

> * The `bdr.node_summary` view, which provides a human readable view of the nodes in the cluster.

Reviewer: ```suggestion
* The `bdr.node_summary` view, which provides a human-readable view of the nodes in the cluster.
```

* The `bdr.local_node` table, which contains information about the local node.
* The uuid values also appear in the naming of the replication slots that are created for each node.

> Although used throughour PGD's node management, the use of UUIDs should not impact any existing functionality or features in PGD. The UUIDs are used internally to identify nodes and groups, and do not change the way that users interact with PGD.

Reviewer: ```suggestion
Although used throughour PGD's node management, the use of UUIDs doesn't affect any existing functionality or features in PGD. The UUIDs are used internally to identify nodes and groups and don't change the way that users interact with PGD.
```

### Why UUIDs?

> UUIDs are used in PGD to provide a unique identifier for each node in the cluster. Previous versions of PGD used the node name as an identifier which could lead to conflicts if two nodes had the same name. By using UUIDs, PGD can ensure that each node has a unique identifier that will not change over time. This is especially important in a distributed system like PGD where nodes may be added or removed from the cluster frequently. The UUID ensures that although a new node may have the same name as an existing node, it will have a different UUID and will not conflict with the existing node.

Reviewer: ```suggestion
UUIDs are used in PGD to provide a unique identifier for each node in the cluster. Previous versions of PGD used the node name as an identifier, which could lead to conflicts if two nodes had the same name. By using UUIDs, PGD can ensure that each node has a unique identifier that will not change over time. This is especially important in a distributed system like PGD where nodes may be added or removed from the cluster frequently. The UUID ensures that although a new node may have the same name as an existing node, it has a different UUID and doesn't conflict with the existing node.
```

### How are UUIDs generated?

When a new node is created, a UUID is generated for that node. This UUID is created using the kernel's strong random number generator and guaranteed to be uniformly random. This ensures that the UUID is unique and cannot be easily guessed. The generated UUID is then stored in the `bdr.node` table and is used to identify the node in the cluster.

### What happens if a node is removed and a replacement added?

If a node is removed from the cluster and a replacement node is added, the replacement node will be assigned a new UUID. This ensures that the replacement node is treated as a separate entity in the cluster and will not conflict with the existing nodes. But, PGD will require that the old node be fully parted from the cluster before it accepts the new node. The UUID of the replacement node will then be used in the same way as the UUIDs of the other nodes in the cluster.

> ### UUID related changes in PGD 6

Reviewer: ```suggestion
### UUID-related changes in PGD 6
```

> * The `generation` field in the `bdr.node` table, which was previously used to differentiate between nodes is no longer used. It will remain at 0 for all nodes.

Reviewer: ```suggestion
* The `generation` field in the `bdr.node` table, which was previously used to differentiate between nodes, is no longer used. It remains at 0 for all nodes.
```

> * the `node_uuid` field in the `bdr.node` table will never be null on PGD6. It may be null in the future with a mixed version cluster.

Reviewer: ```suggestion
* The `node_uuid` field in the `bdr.node` table is never null in PGD 6. It may be null in the future with a mixed version cluster.
```
